org-roam-link.el

```diff
@@ -0,0 +1,341 @@
+;; Boston, MA 02110-1301, USA.
+
+;;; Commentary:
+;;
+;; This library provides the custom roam link type
+;;
+;;; Code:
+;;;; Library Requires
+(require 'org)
+(require 'org-element)
+(require 'dash)
+
+;; org-roam-features
+(require 'org-roam-capture)
+(require 'org-roam-macs)
+
+;; Declarations
+(defvar org-roam-verbose)
+(defvar org-roam-directory)
+(defvar org-roam-buffer--current)
+(declare-function org-roam--in-buffer-p                "org-roam")
+(declare-function org-roam--backlink-to-current-p      "org-roam")
+(declare-function org-roam--org-roam-file-p            "org-roam")
+(declare-function org-roam--org-roam-title-p           "org-roam")
+(declare-function org-roam--get-file-from-title        "org-roam")
+(declare-function org-roam--title-to-slug              "org-roam")
+(declare-function org-roam--format-link-title          "org-roam")
+(declare-function org-roam--format-link                "org-roam")
+(declare-function org-roam--get-title-from-file        "org-roam")
+(declare-function org-roam--get-title-path-completions "org-roam")
+
+;;;; Customizable Variables
+(defcustom org-roam-link-use-roam-links nil
+  "When t `org-roam-insert' inserts roam-link instead of org file-link."
+  :type 'boolean
+  :group 'org-roam)
+
+(defcustom org-roam-link-show-brackets nil
+  "Whether to show brackets around [[roam:]] links."
+  :type 'boolean
+  :group 'org-roam)
+
+;;;; Dynamic variables
+(defvar org-roam-link-message-timer nil
+  "Set by `org-roam-link-show-link-messages' or `org-roam-link-cancel-link-messages'.")
+
+;;;; Custom org-link, roam:
+(defface org-roam-link-brackets
+  '((t :inherit org-link))
+  "Face for roam: link brackets."
+  :group 'org-roam-faces)
+
+(defun org-roam-link--activate (start end _path bracketp)
+  "Hides roam: link prefix and determines additional font-locking.
+Optionally hide brackets before/after the link, or change their face.
+If link uses a Description syntax, hide link path also, and optionally
+show a single bracket pair around Description.
+START and END are buffer position at start/end of the link, PATH is the
+link's path as a string, and BRACKETP is a boolean that is non-nil when the
+link has brackets."
+  (when bracketp
+    (save-excursion
+      (save-match-data
+        (goto-char start)
+        (re-search-forward "\\(\\[\\[\\)\\(roam:\\).*\\(\\]\\]\\)" end t)
+        ;; Optionally hide starting brackets or change their face
+        ;; Can't set invisible with org-roam-link-show-brackets directly
+        (add-text-properties
+         (match-beginning 1)
+         (match-end 1)
+         (if org-roam-link-show-brackets
+             '(face org-roam-link-brackets invisible nil)
+           '(face org-roam-link-brackets invisible t)))
+        (add-text-properties
+         (match-beginning 3)
+         (match-end 3)
+         (if org-roam-link-show-brackets
+             '(face org-roam-link-brackets invisible nil)
+           '(face org-roam-link-brackets invisible t)))
```

Comment:
I'll double-check that it fails, but I remember trying to set `invisible` with a boolean var and it absolutely refused to work.

Reply:
Note that the list is backquoted, so that `org-roam-link-show-brackets' is evaluated in that list.

Similarly you could try something like this in a buffer:
```emacs-lisp
(let ((undelined nil))
  (save-excursion)
  (goto-char (point-max))
  (insert (propertize "test" 'face `(:underline ,undelined))))
```
swapping the value of `underlined` will change the `:undeline` property.

Reply:
"Today I learned ... "

Reply:
It's a very useful feature:
https://www.gnu.org/software/emacs/manual/html_node/elisp/Backquote.html#Backquote

The list splicing syntax in that manual node is handy, too.